addons/mail/static/src/discuss/core/common/discuss_channel_model.js

```diff
@@ -2,17 +2,19 @@ import { fields, Record } from "@mail/core/common/record";
 
 export class DiscussChannel extends Record {
     static _name = "discuss.channel";
+    static _inherits = { "mail.thread": "thread" };
     static id = "id";
 
+    static new() {
+        const channel = super.new(...arguments);
+        // ensure thread is set before reading/writing any other field
+        channel.thread = { id: channel.id, model: "discuss.channel" };
+        return channel;
+    }
```

Comment:
I don't understand this part of the code well, but I wonder if this part is still useful?
It looks like the job of `_inherits`.